webapi/src/main/java/org/sakaiproject/webapi/controllers/AbstractSakaiApiController.java

```diff
@@ -70,4 +76,18 @@ Site checkSite(String siteId) {
             throw new UnknownSiteException(e.getCause());
         }
     }
+
+    protected List<Map<String, String>> getPinnedSiteList() {
+
+        return portalService.getPinnedSites().stream().map(siteId -> {
+
+                try {
+                    Site site = siteService.getSite(siteId);
+                    return Map.of("siteId", siteId, "title", site.getTitle());
+                } catch (IdUnusedException e) {
+                    log.warn("No sites found for siteId {}. Using siteId as the title.", siteId);
+                }
+                return Map.of("siteId", siteId, "title", siteId);
+            }).collect(Collectors.toList());
```

Comment:
this seems strange why not just return an empty list?

Reply:
also you could probably use getOptionalSite

Reply:
@ern Because I'm returning a mapped version of the pinned sites. I don't understand what you mean.

Reply:
@ern getOptionalSite :) Lovely.